orttraining/orttraining/training_api/include/onnxruntime_training_cxx_inline.h

```diff
@@ -208,9 +208,12 @@ inline void CheckpointState::AddProperty(const std::string& property_name, const
     ThrowOnError(GetTrainingApi().AddProperty(p_, property_name.c_str(), OrtPropertyType::OrtFloatProperty, value_p));
   } else if (std::holds_alternative<std::string>(property_value)) {
     std::string value = std::get<std::string>(property_value);
-    auto buffer = std::make_unique<char[]>(value.length() + 1).release();
-    memcpy(buffer, value.c_str(), value.length());
-    ThrowOnError(GetTrainingApi().AddProperty(p_, property_name.c_str(), OrtPropertyType::OrtStringProperty, buffer));
+    auto buffer = std::make_unique<char[]>(value.length() + 1);
```

Comment:
nice catch

Reply:
The test output from a debug build complained about the leak. Seems to have been an issue for a while though.

Reply:
I noticed the same leak complain on my PR recently. Thanks for addressing this. :)